notebooks/tabular_examples/tree_based_models/Python Version of Tree SHAP.ipynb

```diff
@@ -84,45 +93,86 @@
+     "end_time": "2023-10-14T11:34:44.005042883Z",
+     "start_time": "2023-10-14T11:34:42.310798980Z"
+    }
+   }
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
    "source": [
     "### Train XGBoost model"
-   ]
+   ],
+   "metadata": {
+    "collapsed": false
+   }
   },
   {
    "cell_type": "code",
    "execution_count": 4,
-   "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
+      "is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, CategoricalDtype) instead\n",
+      "is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, CategoricalDtype) instead\n",
+      "is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, CategoricalDtype) instead\n"
+     ]
+    }
+   ],
    "source": [
     "bst = xgboost.train(\n",
     "    {\"learning_rate\": 0.01, \"max_depth\": 4}, xgboost.DMatrix(X, label=y), 1000\n",
     ")"
-   ]
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-10-14T11:34:44.807204705Z",
+     "start_time": "2023-10-14T11:34:44.006550044Z"
+    }
+   }
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
    "source": [
     "### Python TreeExplainer\n",
     "\n",
     "This uses numba to speed things up."
-   ]
+   ],
+   "metadata": {
+    "collapsed": false
+   }
   },
   {
    "cell_type": "code",
    "execution_count": 5,
-   "metadata": {},
    "outputs": [],
    "source": [
     "class TreeExplainer:\n",
     "    def __init__(self, model, **kwargs):\n",
-    "        if str(type(model)).endswith(\"sklearn.ensemble.forest.RandomForestRegressor'>\"):\n",
-    "            self.trees = [Tree(e.tree_) for e in model.estimators_]\n",
+    "        if str(type(model)).endswith(\n",
+    "            \"sklearn.ensemble._forest.RandomForestRegressor'>\"\n",
+    "        ):\n",
+    "            # self.trees = [Tree(e.tree_) for e in model.estimators_]\n",
+    "            self.trees = [\n",
+    "                Tree(\n",
+    "                    children_left=e.tree_.children_left,\n",
+    "                    children_right=e.tree_.children_right,\n",
+    "                    children_default=e.tree_.children_right,\n",
+    "                    feature=e.tree_.feature,\n",
+    "                    threshold=e.tree_.threshold,\n",
+    "                    value=e.tree_.value[:, 0, 0],\n",
+    "                    node_sample_weight=e.tree_.weighted_n_node_samples,\n",
+    "                )\n",
+    "                for e in model.estimators_\n",
+    "            ]\n",
     "\n",
     "        # Preallocate space for the unique path data\n",
     "        maxd = np.max([t.max_depth for t in self.trees]) + 2\n",
```

Comment:
Shall we remove the `print(maxd)` line in here? it's causing the 6's to appear in the output which is quite mystifying..

![image](https://github.com/shap/shap/assets/30731072/7c477d5a-31fd-4a4e-867f-f21e1d63e5c2)

Reply:
I agree, I add a commit to remove this.

Reply:
Also corrected the last markdown, since the ratio is closer to 100.